PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs

```diff
@@ -221,6 +221,93 @@ private Boolean CheckIfAppointment(String stringToCheck)
             return stringToCheck.Equals("Appointment");
         }
 
+        /// <summary> This method is calling searchForFirstParameter, searchForOtherParameters to get list of filtered <c>DoctorAppointment</c> of logged patient. </summary>
+        /// /// <param name="dto"><c>AppointmentAdvancedSearchDto</c> is Data Transfer Object of a <c>DoctorAppointment</c> that is be used to filter appointments.
+        /// </param>
+        /// <returns> List of filtered appointments. </returns>
+        public List<DoctorAppointment> AdvancedSearchAppointments(AppointmentAdvancedSearchDto dto)
+        {
+
+            return searchForOtherParameters(GetAppointmentsForPatient(2), dto, searchForFirstParameter(GetAppointmentsForPatient(2), dto));
+
+        }
+
+        /// <summary> This method is getting list of filtered <c>DoctorAppointment</c> that match list of parameters in <c>AppointmentAdvnacedSearchDto</c></summary>
+        /// <param name="appointments"> List of all <c>DoctorAppointment</c> of logged user.
+        /// </param>
+        /// <param name="dto"><c>AppointmentAdvancedSearchDto</c> is Data Transfer Object of a <c>DoctorAppointment</c> that is be used to filter appointments.
+        /// </param>
+        /// <param name="firstAppointments"> List of <c>DoctorAppointment</c> that contains appointments that matches first parameter.
+        /// </param>
+        /// <returns> List of filtered appointments. </returns>
+        private List<DoctorAppointment> searchForOtherParameters(List<DoctorAppointment> appointments, AppointmentAdvancedSearchDto dto, List<DoctorAppointment> firstAppointments)
+        {
+            for (int i = 0; i < dto.RestRoles.Length; i++)
+            {
+                List<DoctorAppointment> othersAppointments = searchForOtherRoles(dto.RestRoles[i], dto.Rest[i], appointments);
+                
+                firstAppointments = searchForLogicOperators(dto.LogicOperators[i], othersAppointments, firstAppointments);
```

Comment:
Write this in the form:
firstAppointments = searchForLogicOperators(dto.LogicOperators[i], searchForOtherRoles(dto.RestRoles[i], dto.Rest[i], appointments), firstAppointments);